server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/UserQuotaRoutesTest.java

```diff
@@ -0,0 +1,457 @@
+            given()
+                .get(QUOTA_USERS + "/" + BOB + "/" + COUNT)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .as(Long.class);
+
+        assertThat(quota).isEqualTo(UNLIMITED);
+    }
+
+    @Test
+    public void getCountShouldReturnStoredValue() throws Exception {
+        int value = 42;
+        maxQuotaManager.setMaxMessage(QuotaRoot.forUser(BOB), value);
+
+        Long actual =
+            given()
+                .get(QUOTA_USERS + "/" + BOB + "/" + COUNT)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .as(Long.class);
+
+        assertThat(actual).isEqualTo(value);
+    }
+
+    @Test
+    public void putCountShouldReturnNotFoundWhenUserDoesntExist() {
+        given()
+            .body("invalid")
+        .when()
+            .put(QUOTA_USERS + "/" + JOE + "/" + COUNT)
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404);
+    }
+
+
+    @Test
+    public void putCountShouldRejectInvalid() throws Exception {
+        Map<String, Object> errors = given()
+            .body("invalid")
+            .put(QUOTA_USERS + "/" + BOB + "/" + COUNT)
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .contentType(ContentType.JSON)
+            .extract()
+            .body()
+            .jsonPath()
+            .getMap(".");
+
+        assertThat(errors)
+            .containsEntry("statusCode", HttpStatus.BAD_REQUEST_400)
+            .containsEntry("type", "InvalidArgument")
+            .containsEntry("message", "Invalid quota. Need to be an integer value greater than 0")
+            .containsEntry("cause", "For input string: \"invalid\"");
+    }
+
+    @Test
+    public void putCountShouldRejectNegative() throws Exception {
+        Map<String, Object> errors = given()
+            .body("-1")
+            .put(QUOTA_USERS + "/" + BOB + "/" + COUNT)
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .contentType(ContentType.JSON)
+            .extract()
+            .body()
+            .jsonPath()
+            .getMap(".");
+
+        assertThat(errors)
+            .containsEntry("statusCode", HttpStatus.BAD_REQUEST_400)
+            .containsEntry("type", "InvalidArgument")
+            .containsEntry("message", "Invalid quota. Need to be an integer value greater than 0");
+    }
+
+    @Test
+    public void putCountShouldAcceptValidValue() throws Exception {
+        given()
+            .body("42")
+            .put(QUOTA_USERS + "/" + BOB + "/" + COUNT)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        assertThat(maxQuotaManager.getMaxMessage(QuotaRoot.forUser(BOB))).isEqualTo(42);
+    }
+
+
+    @Test
+    @Ignore("no link between quota and mailbox for now")
```

Comment:
this test is the same than the previous one (with an indentation issue). Why do you ignore and keep it?

Reply:
there's no way to write the setup part right now because I would like to add more mail data than the quota I try to set.

Reply:
maybe you want to write an integration test for this then?

Reply:
It's overkill IMO because when this will be implemented, Quota will have a link to Mailbox so unit test will do.